src/plugindata.h

```diff
@@ -263,8 +265,66 @@ void plugin_configure_single(GtkWidget *parent);
 void plugin_help(void);
 void plugin_cleanup(void);
 
+/** Called by Geany when a plugin library is loaded.
+ *
+ * This is the original entry point. Implement and export this function to be loadable at all.
+ * Then fill in GeanyPlugin::info and GeanyPlugin::funcs of the passed @p plugin. Finally
+ * GEANY_PLUGIN_REGISTER() and specify a minimum supported API version.
+ *
+ * For all glory details please read @ref howto.
+ *
+ * @param plugin The unique plugin handle to your plugin. You must set some fields here.
+ * @param module The GModule associated with your plugin library.
```

Comment:
> Again, do params to geany_load_module() hurt anyone?

In a way, they "hurt" _everyone_ (who uses the plugin API). If we require unused extra parameters, it just makes people wonder "Oh should I be using that? I wonder what it's for?". IMO we shouldn't stuff extra params into the interface just because we can, we should rather think about the poor user trying to use the API that doesn't know those parameters were added just because they "don't hurt anyone", and in 99% of cases they should be marked with `G_GNUC_UNUSED` or whatever.